src/planwise/component/datasets2.clj

```diff
@@ -197,13 +192,25 @@
   (let [db-spec   (get-db store)
         region-id (:region-id filter-options)
         algorithm (:coverage-algorithm filter-options)
-        options   (:coverage-options filter-options)]
+        options   (:coverage-options filter-options)
+        tags      (str/join "& " (:tags filter-options))]
```

Comment:
Are there any escaping issues with the user-provided tags? You may want to extract the tsquery builder function into some utils namespace and re-use it in the engine component below.